From 8f40c9b006c70326bcfca73daebb54187a65aa5e Mon Sep 17 00:00:00 2001
From: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
Date: Mon, 26 Jun 2017 17:05:01 +0300
Subject: [PATCH] [ecl_containers] Fix gcc v7.1 compilation issue

When built with gcc v7.1 compilation fails with an error message
about using an expression as a function when updating
the private attribute FloatContainerFormatter.prm_width.

The patch uses assignment to update the attribute.

Upstream-Status: Accepted [https://github.com/stonier/ecl_core/pull/58]

Signed-off-by: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
---
 ecl_containers/include/ecl/containers/common/formatters.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ecl_containers/include/ecl/containers/common/formatters.hpp b/ecl_containers/include/ecl/containers/common/formatters.hpp
index e26ec60..5f4c026 100644
--- a/ecl_containers/include/ecl/containers/common/formatters.hpp
+++ b/ecl_containers/include/ecl/containers/common/formatters.hpp
@@ -123,7 +123,7 @@ class ECL_LOCAL FloatContainerFormatter {
          * @param w : the width to use for the inserted float.
          * @return FloatContainerFormatter& : this formatter readied for use with a stream.
          */
-        FloatContainerFormatter& operator()(const unsigned int p, const int w) { format.precision(p); prm_width(w); return *this; } // Permanent
+        FloatContainerFormatter& operator()(const unsigned int p, const int w) { format.precision(p); prm_width = w; return *this; } // Permanent
         /**
          * Convenient stream formatter. This function directly readies the formatter
          * with the specified container and the stored (permanent) settings.
-- 
2.9.3

